partiql-plan/src/main/resources/partiql_plan.ion

```diff
@@ -77,13 +77,21 @@ rex::{
       value: partiql_value,
     },
 
-    var::{
-      ref: int,
-    },
-
-    global::{
-      ref: ref,
-    },
+    var::[
+      // Refers to a value coming from the input record.
+      local::{
+        ref: int
+      },
+      // Refers to a value defined in outer scopes. See https://craftinginterpreters.com/closures.html#upvalues .
+      upvalue::{
+        frameRef: int,
+        valueRef: int,
+      },
+      // Refers to a value in the database environment.
+      global::{
+        ref: ref,
+      },
```

Comment:
```suggestion
      // Refers to a value coming from the input record.
      local::{
        ref: int
      },
      // Refers to a value defined in outer scopes.
      outer::{
        depth: int,
        ref: int,
      },
      // Refers to a value in the database environment.
      global::{
        ref: ref,
      },
```

Our implementation is more like the scope walking in the Java implementation of Lox rather than C implementation hence `depth`. https://craftinginterpreters.com/resolving-and-binding.html#interpreting-resolved-variables We are using the JVM call stack rather than a stack-based VM